postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -440,7 +459,11 @@ def _collect_plan_for_statement(self, row):
                 },
                 'postgres': {k: v for k, v in row.items() if k not in pg_stat_activity_sample_exclude_keys},
             }
-            event['timestamp'] = time.time() * 1000
+            if not obfuscated_plan:
+                # There could be a situation where a 'no plan reason' is provided even when a plan is generated.
+                # This is because some queries in a schema can fail while others succeed. This check
+                # prevents including a 'no plan reason' despite actually having a plan.
+                event['db']['no_plan_reason'] = self._no_explain_reason
```

Comment:
It's not necessary to separate it like this. If the value of the field is `None`/`null`, it won't be stored.

Reply:
Since this is a plan-specific error field would it be better to put it under `db.plan.collection_error`?